runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/DokkaTask.kt

```diff
@@ -57,63 +58,54 @@ open class DokkaTask : DefaultTask(), Configurable {
     var subProjects: List<String> = emptyList()
 
     @Input
-    var impliedPlatforms: MutableList<String> = arrayListOf()
-
     override val pluginsConfiguration: Map<String, String> = mutableMapOf()
 
     @Optional
     @Input
     var cacheRoot: String? = null
 
     @Classpath
-    lateinit var pluginsConfig: Configuration
+    lateinit var pluginsClasspathConfiguration: Configuration
 
     var dokkaSourceSets: NamedDomainObjectContainer<GradlePassConfigurationImpl>
         @Suppress("UNCHECKED_CAST")
         @Nested get() = (DslObject(this).extensions.getByName(SOURCE_SETS_EXTENSION_NAME) as NamedDomainObjectContainer<GradlePassConfigurationImpl>)
         internal set(value) = DslObject(this).extensions.add(SOURCE_SETS_EXTENSION_NAME, value)
 
-    var configuration: GradlePassConfigurationImpl
-        @Suppress("UNCHECKED_CAST")
-        @Nested get() = DslObject(this).extensions.getByType(GradlePassConfigurationImpl::class.java)
-        internal set(value) = DslObject(this).extensions.add(CONFIGURATION_EXTENSION_NAME, value)
-
-    var config: GradleDokkaConfigurationImpl? = null
-
-    // Configure Dokka with closure in Gradle Kotlin DSL
-    fun configuration(action: Action<in GradlePassConfigurationImpl>) = action.execute(configuration)
-
     private val kotlinTasks: List<Task> by lazy {
-        extractKotlinCompileTasks(
-            configuration.collectKotlinTasks ?: { defaultKotlinTasks() })
+        extractKotlinCompileTasks({
```

Comment:
For me: This is not suuuper easy to understand. Can we make this part a little bit easier?

Reply:
Gladly, I'm just not sure how to do it better, considering that the `collectKotlinTasks` have to be a nullable lambda to nullable list of nullable Any, because of Gradle's lifecycle